crates/ruff_workspace/src/configuration.rs

```diff
@@ -454,9 +395,156 @@ impl Configuration {
                 .map(|src| resolve_src(&src, project_root))
                 .transpose()?,
             target_version: options.target_version,
+            typing_modules: options.typing_modules,
+
+            lint: LintConfiguration::from_options(lint, project_root)?,
+
+            format: if let Some(FormatOrOutputFormat::Format(format)) = options.format {
+                Some(FormatConfiguration::from_options(format)?)
+            } else {
+                None
+            },
+        })
+    }
+
+    #[must_use]
+    pub fn combine(self, config: Self) -> Self {
```

Comment:
that's something i'd like a macro for (not a review comment)

Reply:
Potentially, but then, macros are hard to understand and its not so much trouble to write it.